compiler/rustc_incremental/src/persist/fs.rs

```diff
@@ -664,6 +676,7 @@ fn is_old_enough_to_be_collected(timestamp: SystemTime) -> bool {
     timestamp < SystemTime::now() - Duration::from_secs(10)
 }
 
+/// Runs garbage collection for the current session.
```

Comment:
It garbage collects all incremental session directories **except** the current session, right?

Reply:
Good question. I'll double check.

Reply:
It looks like this operates on a single session only.

Reply:
The garbage collection logic is complex enough here that I wouldn't feel comfortable discussing its behavior in the public doc comment without input from an expert.